pkg/generators/servers.go

```diff
@@ -809,3 +841,22 @@ func (g *ServersGenerator) mapMethodNameToHTTPMethod(name *names.Name) string {
 		return http.MethodGet
 	}
 }
+
+func (g *ServersGenerator) readerName(typ *concepts.Type) string {
+	// see helpers.go file where the following methods are defined.
+	version := typ.Owner()
+	switch {
+	case typ == version.Integer():
+		return "ParseInteger"
+	case typ == version.Float():
+		return "ParseFloat"
+	case typ == version.String():
+		return "ParseString"
+	case typ == version.Date():
+		return "ParseDate"
+	case typ == version.Boolean():
+		return "ParseBoolean"
+	default:
+		return "ParseString"
```

Comment:
This will cause a compiler error if one day we add a new built-in type, or if we forget one in this switch. I think it is better to report that we don't know how to handle that type.

Reply:
No sure that I follow what you mean.